renderer/test/js_ethereum_provider_browsertest.cc

```diff
@@ -229,14 +283,74 @@ IN_PROC_BROWSER_TEST_F(JSEthereumProviderBrowserTest, AttachIfWalletCreated) {
   const GURL url = https_server_.GetURL("/simple.html");
   ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
 
-  constexpr char kEvalIsBraveWallet[] = "window.ethereum.isBraveWallet";
+  {
+    constexpr char kEvalIsBraveWallet[] = "window.ethereum.isBraveWallet";
+    EXPECT_TRUE(content::EvalJs(primary_main_frame(), kEvalIsBraveWallet)
+                    .ExtractBool());
+  }
+
+  {
+    constexpr char kEvalIsBraveWallet[] = "window.braveEthereum.isBraveWallet";
+    EXPECT_TRUE(content::EvalJs(primary_main_frame(), kEvalIsBraveWallet)
+                    .ExtractBool());
+  }
+
+  EXPECT_EQ(browser()->tab_strip_model()->GetTabCount(), 1);
+}
+
+IN_PROC_BROWSER_TEST_F(JSEthereumProviderBrowserTest, EIP6369) {
+  auto* keyring_service =
+      brave_wallet::KeyringServiceFactory::GetServiceForContext(
+          browser()->profile());
+  keyring_service->CreateWallet("password", base::DoNothing());
+
+  brave_wallet::SetDefaultEthereumWallet(
+      browser()->profile()->GetPrefs(),
+      brave_wallet::mojom::DefaultWallet::BraveWalletPreferExtension);
+
+  const GURL url = https_server_.GetURL("/simple.html");
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
+
+  EXPECT_TRUE(
+      content::EvalJs(primary_main_frame(), kTestEIP6963).ExtractBool());
+
+  EXPECT_EQ(browser()->tab_strip_model()->GetTabCount(), 1);
+}
+
+IN_PROC_BROWSER_TEST_F(JSEthereumProviderBrowserTest,
+                       EIP6369_WithoutEthereumInstalled) {
```

Comment:
This test suite should be `EIP6369_MetaMaskAttached`

Reply:
Done